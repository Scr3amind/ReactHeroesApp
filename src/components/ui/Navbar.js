import React, { useContext } from 'react'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {
    const history = useHistory()
    const {user, dispatch} = useContext(AuthContext);
    const {name} = user;
    const handleLogout = () => {
        dispatch({
            type: types.logout,
            
        })

        history.replace('/login');
    }

    return (
        <nav className="bg-gradient-to-r from-gray-700 to-black shadow-2xl relative z-10 py-4 animate__animated animate__fadeInDown">
            <div className="container mx-auto">
                <div className="sm:flex justify-around">
                    {/* Site title */}
                    <Link to="/" className="text-white text-3xl font-bold p-3">
                        Heroes App
                    </Link>
                    {/* Items */}
                    <ul className="text-gray-400 sm:self-center text-xl border-t sm:border-none">
                        <li className="sm:inline">
                            <NavLink exact to="/marvel" activeClassName="text-white font-bold" className="p-3 hover:text-white">
                                Marvel
                            </NavLink>
                        </li>
                        <li className="sm:inline">
                            <NavLink exact to="/dc" activeClassName="text-white font-bold" className="p-3 hover:text-white">
                                DC
                            </NavLink>
                        </li>
                        <li className="sm:inline">
                            <NavLink exact to="/search" activeClassName="text-white font-bold" className="p-3 hover:text-white">Search</NavLink>
                        </li>
                        <li className="sm:inline">
                            <button className="p-3 hover:text-white" onClick={handleLogout}>
                                Logout
                            </button>
                        </li>
                        <li className="sm:inline">
                            <span className="px-3 text-gray-700 bg-white rounded-lg bg-opacity-80 font-bold">{name}</span>
                        </li>

                    </ul>

                </div>
            </div>

        </nav>
    )
}
