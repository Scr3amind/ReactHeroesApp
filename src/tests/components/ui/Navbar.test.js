import { mount } from "enzyme"
import { Navbar } from "../../../components/ui/Navbar"
import { AuthContext } from "../../../auth/AuthContext";
import '../../../setupTest'
import { MemoryRouter, Router } from "react-router";
import { types } from "../../../types/types";

describe('Pruebas en <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
    }

    const contextValue = {
        dispatch : jest.fn(),
        user: {
            logged: true,
            name: 'Isaac'
        }
    }

    afterEach(() => {
        jest.clearAllMocks();
    });

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar el campo name que contiene el context', () => {
        expect(wrapper.html().includes(contextValue.user.name)).toBe(true);
    })

    test('debe de llamar el logout y usar history', () => {
        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalledWith(
            {
                type:types.logout
            }
        );

        expect(historyMock.replace).toHaveBeenCalledWith('/login');
        
    })
    
    
    

    
})
