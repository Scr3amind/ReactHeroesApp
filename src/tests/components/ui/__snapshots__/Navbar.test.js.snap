// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> debe de mostrarse correctamente 1`] = `
<nav
  className="bg-gradient-to-r from-gray-700 to-black shadow-2xl relative z-10 py-4 animate__animated animate__fadeInDown"
>
  <div
    className="container mx-auto"
  >
    <div
      className="sm:flex justify-around"
    >
      <a
        className="text-white text-3xl font-bold p-3"
        onClick={[Function]}
      >
        Heroes App
      </a>
      <ul
        className="text-gray-400 sm:self-center text-xl border-t sm:border-none"
      >
        <li
          className="sm:inline"
        >
          <a
            aria-current={null}
            className="p-3 hover:text-white"
            onClick={[Function]}
          >
            Marvel
          </a>
        </li>
        <li
          className="sm:inline"
        >
          <a
            aria-current={null}
            className="p-3 hover:text-white"
            onClick={[Function]}
          >
            DC
          </a>
        </li>
        <li
          className="sm:inline"
        >
          <a
            aria-current={null}
            className="p-3 hover:text-white"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
        <li
          className="sm:inline"
        >
          <button
            className="p-3 hover:text-white"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
        <li
          className="sm:inline"
        >
          <span
            className="px-3 text-gray-700 bg-white rounded-lg bg-opacity-80 font-bold"
          >
            Isaac
          </span>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
